traffic_ops/testing/api/v4/tenants_test.go

```diff
@@ -16,546 +16,323 @@ package v4
-		t.Errorf("tenant4user got delivery service %s with tenant1, expected: no ds", ExtractXMLID(&ds))
-	}
-
-	setTenantActive(t, "tenant3", false)
-	opts.QueryParameters.Set("xmlId", "ds3")
-	resp, _, err = tenant3Session.GetDeliveryServices(opts)
-	if err != nil {
-		t.Errorf("Unexpected error fetching Delivery Services filtered by XMLID 'ds3': %v - alerts: %+v", err, resp.Alerts)
-	}
-	for _, ds := range resp.Response {
-		t.Errorf("tenant3user was inactive, but got delivery service %s with tenant3, expected: no ds", ExtractXMLID(&ds))
-	}
-
-	for _, tn := range originalTenants.Response {
-		if tn.Name == "root" {
-			continue
-		}
-		if resp, _, err := TOSession.UpdateTenant(tn.ID, tn, client.RequestOptions{}); err != nil {
-			t.Fatalf("restoring original tenants: %v - alerts: %+v", err, resp.Alerts)
+func validateTenantPagination(paginationParam string) utils.CkReqFunc {
+	return func(t *testing.T, _ toclientlib.ReqInf, resp interface{}, _ tc.Alerts, _ error) {
+		paginationResp := resp.([]tc.Tenant)
+
+		opts := client.NewRequestOptions()
+		opts.QueryParameters.Set("orderby", "id")
+		respBase, _, err := TOSession.GetTenants(opts)
+		assert.RequireNoError(t, err, "Cannot get Tenants: %v - alerts: %+v", err, respBase.Alerts)
+
+		tenants := respBase.Response
+		assert.RequireGreaterOrEqual(t, len(tenants), 3, "Need at least 3 Tenants in Traffic Ops to test pagination support, found: %d", len(tenants))
+		switch paginationParam {
+		case "limit:":
+			assert.Exactly(t, tenants[:1], paginationResp, "expected GET Tenants with limit = 1 to return first result")
+		case "offset":
+			assert.Exactly(t, tenants[1:2], paginationResp, "expected GET Tenants with limit = 1, offset = 1 to return second result")
+		case "page":
+			assert.Exactly(t, tenants[1:2], paginationResp, "expected GET Tenants with limit = 1, page = 2 to return second result")
 		}
 	}
 }
 
-func setTenantActive(t *testing.T, name string, active bool) {
-	opts := client.NewRequestOptions()
-	opts.QueryParameters.Set("name", name)
-	resp, _, err := TOSession.GetTenants(opts)
-	if err != nil {
-		t.Fatalf("cannot get Tenants filtered by name '%s': %v - alerts: %+v", name, err, resp.Alerts)
-	}
-	if len(resp.Response) != 1 {
-		t.Fatalf("Expected exactly one Tenant to exist with the name '%s', found: %d", name, len(resp.Response))
-	}
-	tn := resp.Response[0]
-
-	tn.Active = active
-	response, _, err := TOSession.UpdateTenant(tn.ID, tn, client.RequestOptions{})
-	if err != nil {
-		t.Fatalf("cannot update Tenant: %v - alerts: %+v", err, response.Alerts)
+func GetTenantID(t *testing.T, name string) func() int {
+	return func() int {
+		opts := client.NewRequestOptions()
+		opts.QueryParameters.Set("name", name)
+		tenants, _, err := TOSession.GetTenants(opts)
+		assert.RequireNoError(t, err, "Get Tenants Request failed with error:", err)
+		assert.RequireEqual(t, 1, len(tenants.Response), "Expected response object length 1, but got %d", len(tenants.Response))
```

Comment:
If `len(tenants.Response)` is not 1 for some reason, does this stop the test here? Or does it still proceed to return `tenants.Response[0].ID`? If it is the former, then we're fine. If not, then the `return` statement could cause a panic.

Reply:
Yes it'll stop the test there. Require* functions call t.Fatal which stops test execution.